src/pip/_internal/cache.py

```diff
@@ -102,6 +106,10 @@ def _link_for_candidate(self, link, candidate):
 
         return index.Link(path_to_url(path))
 
+    def cleanup(self):
```

Comment:
This should be on `WheelCache`. Also, since it only clears the ephemeral cache, I'd suggest renaming this to `cleanup_ephem`.

Reply:
After thinking about this some more, I don't agree. I think that first of all, `cleanup` methods have very little support in the Python ecosystem. Second of all, when there is a `cleanup` method, it should be defined on the base class so that it's always called after using the object and specific implementations should go on derived classes.

Reply:
So, specifically, I propose making this `pass`, and then moving the guts to `WheelCache`.